bn.c

```diff
@@ -273,6 +273,45 @@ void bignum_mul(struct bn* a, struct bn* b, struct bn* c)
 }
 
 
+void bignum_mul_alt(struct bn *a, struct bn *b, struct bn *c)
+{
+  require(a, "a is null");
+  require(b, "b is null");
+  require(c, "c is null");
+
+  bignum_init(c);
+  DTYPE_TMP tmp = 0;
+  DTYPE tmp_to_add = 0;
+
+  int usable_len = BN_ARRAY_SIZE;
+
+  /*  this section speads up algorithm by "cutting" len of bignum*/
```

Comment:
I don't want to pull this part. Let me first comment it, and then tell you the reason why I would like to avoid it,

I appreciate the effort and understand why you would want this here (and in every other functions. You could also maintain a bitset of highest used bit to optimize for speed.
However it adds to the size of both source- and object code. I also fear you might miss out on vectorization optimizations from the compiler, but I wouldn't know for sure. It also makes you susceptible to leaking info via timing if used in encryption algorithms. I mostly want to take a raincheck because it also breaks the beautiful simplicity of the naïive algorithm.

Thanks for the PR - I really appreciate the effort you've put into this

Reply:
Vector operations bases on processor words so don't worry.
I understand the timing problem in cryptography.
I am not sure are you talking about the same peace of code.

Reply:
So the section with finding actual length of number can be removed.